### PR TITLE
feat/nice_date utils

### PR DESCRIPTION
**Pull request by JarbasAl**
https://github.com/MycroftAI/lingua-franca/issues/198

### CODE REVIEW

File: lingua_franca/format.py [hunk 1]
```diff
@@ -370,6 +374,39 @@ def nice_date_time(dt, lang='', now=None, use_24hour=False,
                                              use_ampm)
 
 
+@localized_function(run_own_code_on=[FunctionNotLocalizedError])
+def nice_day(dt, date_format='MDY', include_month=True, lang=""):
```
Comment (NeonDaniel) [inline lingua_franca/format.py:378]: perhaps `date_format` should be an enum since there are only 3(?) valid options right now (MDY, DMY, YMD).

**Review by NeonDaniel — CHANGES_REQUESTED** [review]
Added functions need docstrings. If we're not using a standard date formatting string, valid options should either be tied to an enum class or documented in docstrings; unsupported formats should probably raise an exception

File: lingua_franca/format.py (hunk 1 above)
Comment (JarbasAl) [reply lingua_franca/format.py:378]: can't hurt, but note this comes from mycroft.conf and the valid options are basically defined there, so let's define what the enum should contain clearly

File: lingua_franca/format.py [hunk 2]
```diff
@@ -370,6 +374,39 @@ def nice_date_time(dt, lang='', now=None, use_24hour=False,
...
+    if include_month:
+        month = nice_month(dt, date_format, lang)
+        if date_format == 'MDY':
+            return "{} {}".format(month, dt.strftime("%d"))
+        else:
+            return "{} {}".format(dt.strftime("%d"), month)
+    return dt.strftime("%d")
```
Comment (emphasize) [inline lingua_franca/format.py:385]: just realized that a one digit day is formatted with "0x". I don't know if all TTS can handle that.
Better `dt.strftime("%-d")`

File: lingua_franca/format.py [hunk 3]
```diff
@@ -392,6 +429,34 @@ def nice_year(dt, lang='', bc=False):
...
+    if date_format == 'MDY':
+        dtstr = dt.strftime("%-m/%-d/%Y")
+    elif date_format == 'DMY':
+        dtstr = dt.strftime("%d/%-m/%-Y")
+    elif date_format == 'YMD':
+        dtstr = dt.strftime("%Y/%-m/%-d")
+    else:
+        raise ValueError("invalid date_format")
+    return {
+        "date_string": dtstr,
```
Comment (emphasize) [inline lingua_franca/format.py:451]: I feel we should include abbreviations coming from
`nice_month`/`nice_weekday`with a changed signature
```python
def nice_month(dt, abbreviation=False, lang=""):
```
in both cases date_format is irrelevant

File: lingua_franca/format.py [hunk 4]
```diff
@@ -370,6 +374,39 @@ def nice_date_time(dt, lang='', now=None, use_24hour=False,
...
+        else:
+            return "{} {}".format(dt.strftime("%d"), month)
+    return dt.strftime("%d")
+
+
+@localized_function(run_own_code_on=[FunctionNotLocalizedError])
+def nice_weekday(dt, lang=""):
+    if lang in date_time_format.lang_config.keys():
+        localized_day_names = list(
+            date_time_format.lang_config[lang]['weekday'].values())
```
Comment (emphasize) [inline lingua_franca/format.py:392]: There has to be a `cache` call upon `date_time_format` to be safe to say that the keys are there.

I'm inclined to rewrite the class, that poor class is crammed
object dict created on init, with a helper `get_date_time_format(lang)`

**Comment by JarbasAl** [comment]
I'm merging this one and addressing feedback later as this is a direct port of utils used in the datetime skill, 

I want to merge the homescreen PR needing this to reduce the insane amount of log spam it produces, right now im focused on getting 0.0.7 core release out and dont have time to focus on LF

**Pull request merged**